### PR TITLE
Class consolidation

### PR DESCRIPTION
**Pull request by ghost**
This removes the goal and system_state classes and their associated tests and instead uses the system class as a state representation within the genome.

From one perspective this is overkill, since the system class contains more than is needed to represent an upgrade step, but the architectural simplification may help to make the program simpler to understand.

### CODE REVIEW

File: src/system.py [hunk 1]
```diff
@@ -37,134 +37,135 @@ class System:
...
             logger.debug("Creating System from list")
             self.programs=programs
-        
+
         # Empty system, expected to use addProgram() after creation
         else:
             logger.debug("Creating an empty System")
             self.programs=[]
-    
-        logger.debug("Program list for this system: %s"%list2str(self.programs))
```
Comment (leeming) [inline src/system.py:78]: Unless python3 has corrected this, printing a list (tostring) is not recursive and will just return [<Program>, <Program>, ...]

I will need to double check this

Comment (ghost) [inline src/system.py:78]: In this case does the list of programs need to be recursive?

Comment (leeming) [inline src/system.py:78]: see https://paste.codethink.co.uk/?713 for example of what list2str does

Comment (ghost) [inline src/system.py:78]: does that work with python3?

Comment (leeming) [inline src/system.py:78]: Python 3.4.2

```
python3 example.py 
[<__main__.Foo object at 0x7f2d5bfe75c0>, <__main__.Foo object at 0x7f2d5bfe75f8>]
[bar, foo]
```

Python 2.7

```
python example.py 
[<__main__.Foo instance at 0x7fe498ebb758>, <__main__.Foo instance at 0x7fe498ebb7a0>]
[bar, foo]
```

**Pull request closed without merging**